--- Makefile	2006-08-26 00:56:48.000000000 +0200
+++ Makefile.oden	2008-12-21 23:56:42.000000000 +0100
@@ -4,13 +4,15 @@
 SRC = $(subst .o,.c,$(OBJ))
 
 CFLAGS += -g -Wall
-LDFLAGS += -lpthread
+LIBS += -lpthread
+LDFLAGS += 
+
 CC = gcc
 
 all: $(BINARY)
 
 prcsys: $(OBJ)
-	$(CC) $(CFLAGS) -o prcsys $(OBJ) $(LDFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o prcsys $(OBJ) $(LIBS)
 
 %.o: %.c
 	$(CC) $(CFLAGS) -c $< -o $@
